File: src/test/java/com/example/probation/service/impl/RoleServiceImplTest.java
package com.example.probation.service.impl;

import com.example.probation.core.entity.Role;
import com.example.probation.repository.RoleRepository;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.Mockito;
import org.mockito.junit.jupiter.MockitoExtension;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.mockito.ArgumentMatchers.any;

@ExtendWith(MockitoExtension.class)
class RoleServiceImplTest {
    @Mock
    private RoleRepository repository;
    @InjectMocks
    private RoleServiceImpl service;

    @Test
    void getROleByRole() {
        final Role role = new Role();
        final String newRole = "USER";
        role.setRole(newRole);
        final String userRole = "USER";
        Mockito.when(repository.getRoleByRole(any())).thenReturn(role);
        assertEquals(service.getRoleByRole(userRole).getRole(), role.getRole());
    }
}

File: src/test/kotlin/com/example/probation/service/RoleServiceImplTest.kt
package com.example.probation.service

import com.example.probation.core.entity.Role
import com.example.probation.core.enums.Roles
import com.example.probation.repository.RoleRepository
import com.example.probation.service.impl.RoleServiceImpl
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.mockito.ArgumentMatchers.any
import org.mockito.InjectMocks
import org.mockito.Mock
import org.mockito.Mockito
import org.mockito.junit.jupiter.MockitoExtension

@ExtendWith(MockitoExtension::class)
class RoleServiceImplTest {
    @Mock
    lateinit var repository: RoleRepository
    @InjectMocks
    lateinit var service: RoleServiceImpl

    @Test
    fun getRoleByRoleName() {
        val userRole = Roles.USER.role
        val role = Role(userRole)
        Mockito.`when`(repository.getRoleByRoleName(any(String::class.java) ?: String())).thenReturn(role)
        Assertions.assertEquals(service.getRoleByRoleName(userRole).roleName, role.roleName)
    }
}
